### PR TITLE
DP-33158: Caching issues with menu local tasks

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-****

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_admin_toolbar/src/Plugin/Block/MessageTabs.php [hunk 1]
```diff
@@ -64,9 +64,12 @@ public function build() {
     $cacheability = new CacheableMetadata();
     $primary = $this->localTaskManager->getLocalTasks($this->routeMatch->getRouteName(), 0);
     $secondary = $this->localTaskManager->getLocalTasks($this->routeMatch->getRouteName(), 1);
+    $local_tasks = CacheableMetadata::createFromObject($this->localTaskManager);
+
     $cacheability = $cacheability
       ->merge($primary['cacheability'])
-      ->merge($secondary['cacheability']);
+      ->merge($secondary['cacheability'])
+      ->merge($local_tasks);
```
Comment (weitzman) [inline docroot/modules/custom/mass_admin_toolbar/src/Plugin/Block/MessageTabs.php:72]: Nice find ... I think you can replace these lines with (untested):

```
$cacheability
  ->addCacheableDependency($primary)
  ->addCacheableDependency($secondary)
  ->addCacheableDependency($this->localtaskManager);
```

Comment (arthurbaghdas) [reply docroot/modules/custom/mass_admin_toolbar/src/Plugin/Block/MessageTabs.php:72]: This is the one with the suggested change
Screenshot 2024-05-15 at 18 10 19
The one below is  with the change suggested in the PR. as you can see if we change this we will lose some cache tags and the route cache context
Screenshot 2024-05-15 at 18 10 01

**Comment by tom-fleming** [comment]
@arthurbaghdas I copied these changes over to #2515 . Could we keep this PR open for now but in draft?

**Comment by tom-fleming** [comment]
Closing since changes moved to hofix in #2515

**Pull request closed without merging**